routines/analyze.go

```diff
@@ -0,0 +1,103 @@
+package routines
+
+import (
+	"log"
+	"time"
+
+	"github.com/csivitu/bl0b/ctftime"
+
+	"github.com/csivitu/bl0b/db"
+)
+
+func computeStatus(start time.Time, end time.Time) string {
+	t := time.Now().Unix()
+	startUnix := start.Unix()
+	endUnix := end.Unix()
+
+	if startUnix <= t && t < endUnix {
+		return "ongoing"
+	}
+
+	if t < startUnix && t < endUnix {
+		return "upcoming"
+	}
+
+	return "over"
+}
+
+func handleUpcoming(event *ctftime.Event, DB *db.Database) {
+	if event.Status == "upcoming" {
+		return
+	}
+
+	err := DB.ModifyEventStatus(event.ID, "upcoming")
+	if err != nil {
+		log.Println("Error while modifying event " + event.Title + " in handleUpcoming")
+	}
+}
+
+func handleOngoing(event *ctftime.Event, DB *db.Database) {
+	// If the status is already ongoing, return
+	if event.Status == "ongoing" {
+		return
+	}
+
+	err := DB.ModifyEventStatus(event.ID, "ongoing")
+	if err != nil {
+		log.Println("Error while modifying event " + event.Title + " in handleOngoing")
+		log.Println(err)
+	}
+}
+
+func handleOver(event *ctftime.Event, DB *db.Database) {
+	err := DB.DeleteEventByID(event.ID)
+	if err != nil {
+		log.Println("Error while deleting event " + event.Title)
+		log.Println(err)
+	}
+}
+
+func analyze() {
+	DB := db.New()
+	defer DB.Close()
+
+	events, err := DB.GetEvents()
+	if err != nil {
+		log.Println("Analyze could not get events from database!")
+		panic(err)
+	}
+
+	for _, event := range events {
+		status := computeStatus(event.Start, event.Finish)
+
+		switch status {
+		case "upcoming":
```

Comment:
I think its a better idea both database wise (smaller field size), and code qualitiy wise to make an `enum` to represent each of these states (and then use that everywhere for comparisions, and function signatures), and store in the DB as an integer field.
So the enum could be something like: 
```go
type Status int
const (
        upcoming Status = iota
        ongoing
        over
)
```

and now you can use `upcoming`, `ongoing`, `over` enums instead of strings everywhere in the code, and if a function needs to take a status as an arg, it can use the `Status` type.

Reply:
So, there would be a number representing every state? Right now there are just 3 states, but there would be more later on. In that case, I think strings wouldn't be a bad idea too because you know the status even when you look at the DB. We can however make consts with variable names for each of the states like `const state string`, `state = "state"`...

Reply:
The enum constants can be strings/numbers any yeah, but the main thing I meant was define a Status type can use that everywhere in your code, instead of repeating constants everywhere

Reply:
Ah okay, will do that.

Reply:
Yes